server/models/Board.js

```diff
@@ -1,29 +1,108 @@
 const mongoose = require("mongoose");
+const Column = require("../models/Column");
+const Card = require("../models/Card");
 
-const boardSchema = new mongoose.Schema({
-  name: {
-    type: String,
-    required: true,
+const boardSchema = new mongoose.Schema(
+  {
+    name: {
+      type: String,
+      required: true,
+    },
+    columns: [
+      { type: mongoose.Schema.Types.ObjectId, ref: "column", required: false },
+    ],
+    user: {
+      type: mongoose.Schema.Types.ObjectId,
+      ref: "user",
+      required: true,
+    },
   },
-  columns: {
-    type: Array,
-    required: false,
-  },
-  user: {
-    type: mongoose.Types.ObjectId,
-    ref: "user",
-    required: true,
-  },
-  createdAt: {
-    type: Date,
-    default: Date.now,
-  },
-});
+  { timestamps: true }
+);
 
+// Method that will remove the user's password before returning the board; this can be redone to ensure we don't pop user
 boardSchema.methods.removePassword = function () {
-  const object = this.toObject();
-  delete object.user.password;
-  return object;
+  const board = this.toObject();
+  delete board.user.password;
+  return board;
+};
```

Comment:
When we are populating a user on the board just deselect their password with `.select("-password")` by doing this we utilize Mongoose more and remove the unnecessary methods for removing password across multiple models.

Reply:
I have done this in my PR but feel free to go a head and set the password property on the user to `select: false`